Userland/Applications/SoundPlayer/SoundPlayerWidgetAdvancedView.cpp

```diff
@@ -302,9 +305,17 @@ void SoundPlayerWidgetAdvancedView::try_fill_missing_info(Vector<M3UEntry>& entr
         if (!entry.extended_info->track_display_title.has_value())
             entry.extended_info->track_display_title = LexicalPath(entry.path).title();
         if (!entry.extended_info->track_length_in_seconds.has_value()) {
-            if (entry_path.has_extension("wav")) {
-                auto wav_reader = Audio::Loader::create(entry.path);
-                entry.extended_info->track_length_in_seconds = wav_reader->total_samples() / wav_reader->sample_rate();
+            auto file_extension = entry_path.extension();
+            bool is_supported = false;
+            for (const auto& ext : supported_file_extensions) {
+                if (file_extension == ext) {
+                    is_supported = true;
+                    break;
+                }
+            }
+            if (is_supported) {
+                auto audio_reader = Audio::Loader::create(entry.path);
+                entry.extended_info->track_length_in_seconds = audio_reader->total_samples() / audio_reader->sample_rate();
             }
```

Comment:
Why not let `Audio::Loader` decide if it supports the file instead of deciding that here?
```suggestion
            if (auto reader = Audio::Loader::create(entry.path); !reader.has_error())
                entry.extended_info->track_length_in_seconds = reader->total_samples() / reader->sample_rate();
```

Reply:
I hadn't considered that option, sounds good